packages/reactivity/src/effect.ts

```diff
@@ -234,9 +234,17 @@ export class ReactiveEffect<T = any>
 
 let batchDepth = 0
 let batchedSub: Subscriber | undefined
+const batchedCleanup: (() => void)[] = []
```

Comment:
The current implementation seems to be going out of its way to avoid arrays, instead using a linked list via the `next` property. I'm not entirely sure of the full rationale behind that decision (I'd assume performance), but introducing an array here seems to go against that design choice.

Reply:
updated